### PR TITLE
Support Azure Key Vault

### PR DESCRIPTION
**Pull request by glahaye**
### Motivation and Context
We need a way to store secrets used by the server that works in Azure

### Description
Add KV support in configuration system

### Contribution Checklist
- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs [hunk 1]
```diff
@@ -20,7 +21,14 @@ public static IHostBuilder ConfigureAppSettings(this IHostBuilder host)
             builder.AddJsonFile($"appsettings.{environment}.json", true, true);
             builder.AddEnvironmentVariables();
             builder.AddUserSecrets(Assembly.GetExecutingAssembly(), optional: true, reloadOnChange: true);
-            // For settings from Key Vault, see https://learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-7.0
+            // For settings from Key Vault, see https://learn.microsoft.com/en-us/aspnet/core/security/key-vault-configuration?view=aspnetcore-8.0
+            string? keyVaultName = ctx.Configuration["KeyVaultName"];
+            if (!string.IsNullOrEmpty(keyVaultName))
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs:26]: [nit] Recommend `IsNullOrWhitespace`

File: samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs [hunk 2]
```diff
@@ -20,7 +21,14 @@ public static IHostBuilder ConfigureAppSettings(this IHostBuilder host)
...
+            {
+                builder.AddAzureKeyVault(
+                    new Uri($"https://{keyVaultName}.vault.azure.net/"),
+                    new DefaultAzureCredential());
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs:30]: DefaultAzureCredential is a *beast*. Please add a reference to its documentation to keep the insanity meter from filling up. https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json [hunk 3]
```diff
@@ -29,9 +29,11 @@
        Ex: dotnet user-secrets set "EmbeddingConfig:Key" "MY_EMBEDDING_KEY" */
   },
 
-  // Directory from which to load semantic skills
+  // Directory from which to load semantic skills, if used
   "SemanticSkillsDirectory": "", // Ex: ".\\SemanticSkills" (relative to executable)
 
+  "KeyVaultName": "", // Name of the Key Vault for secrets, if used - Ex: "myvault"
```
Comment (adrianwyatt) [inline samples/apps/copilot-chat-app/SKWebApi/appsettings.json:35]: I recommend changing this to KeyVaultURI. When we get outside of the Azure Public cloud, the URIs change (e.g., government and sovereign clouds).

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs (hunk 1 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs:26]: Changed.

File: samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs (hunk 2 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/Config/ConfigExtensions.cs:30]: Sure

File: samples/apps/copilot-chat-app/SKWebApi/appsettings.json (hunk 3 above)
Comment (glahaye) [reply samples/apps/copilot-chat-app/SKWebApi/appsettings.json:35]: Doing so!

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**